cmake/findDependencies.cmake

```diff
@@ -47,7 +47,11 @@ endif()
 
 if (NOT USE_BUNDLED_TINYXML2)
     find_package(tinyxml2 QUIET)
-    if (NOT tinyxml2_FOUND)
+    if (tinyxml2_FOUND)
+        if (NOT tinyxml2_LIBRARIES)
+            set(tinyxml2_LIBRARIES "tinyxml2")
```

Comment:
This is wrong. This line passes `tinyxml2` to `target_link_libraries`, but since `tinyxml2` isn't a CMake target, it will be passed as a `-l` flag to the link step.

This means that if a user sets `tinyxml2_ROOT` to a newer upstream version of `tinyxml2` or to a patched copy, the _actual_ library will be ignored in favor of one on the library path. Using the CMake target `tinyxml2::tinyxml2` would fix this.

Reply:
ATTN @gdsotirov

Reply:
@alexreinking I saw your response even without the follow up ATTN, but this MR is not mine...

Reply:
Good to know. Since both you and @c72578 weighed in on the tinyxml2 issue, I (incorrectly) assumed you were working together on this PR.

Reply:
Well, I'm trying to help, but I'm simply not sure what is the correct solution after the exchanges... If you look at [cmake/findDependencies.cmake](../blob/master/cmake/findDependencies.cmake) they detect Z3 library in the same way.

Reply:
Here's a sketch that would be better.

```
find_package(tinyxml2 QUIET)
if (TARGET tinyxml2::tinyxml2)
  set(tinyxml2_LIBRARIES "tinyxml2::tinyxml2"
      CACHE INTERNAL "tinyxml2 libraries")
else ()
  # as before ...
endif ()
```

Reply:
> Here's a sketch that would be _better_.
> 
> ```
> find_package(tinyxml2 QUIET)
> if (TARGET tinyxml2::tinyxml2)
>   set(tinyxml2_LIBRARIES "tinyxml2::tinyxml2")
> else ()
>   # as before ...
> endif ()
> ```

@alexreinking Thanks for taking the time to have a look at this PR and the feedback.
I have updated the PR according to your suggestion.